server/src/internalClusterTest/java/org/opensearch/search/SearchWeightedRoutingIT.java

```diff
@@ -871,6 +887,42 @@ public void testPreferenceSearchWithWeightedRouting() {
         assertEquals(RestStatus.OK.getStatus(), searchResponse.status().getStatus());
     }
 
+    /**
+     * Assert that preference based search with preference type is not allowed with strict weighted shard routing
+     */
+    public void testStrictWeightedRouting() {
+
+        Settings commonSettings = Settings.builder()
+            .put("cluster.routing.allocation.awareness.attributes", "zone")
+            .put("cluster.routing.allocation.awareness.force.zone.values", "a,b,c")
+            .put("cluster.routing.weighted.fail_open", true)
+            .put("cluster.routing.weighted.strict", true)
+            .build();
+
+        int nodeCountPerAZ = 1;
+        Map<String, List<String>> nodeMap = setupCluster(nodeCountPerAZ, commonSettings);
+
+        int numShards = 10;
+        int numReplicas = 1;
+        setUpIndexing(numShards, numReplicas);
+
+        logger.info("--> setting shard routing weights for weighted round robin");
+        Map<String, Double> weights = Map.of("a", 1.0, "b", 1.0, "c", 0.0);
+        setShardRoutingWeights(weights);
+        String nodeInZoneA = nodeMap.get("a").get(0);
+        String customPreference = randomAlphaOfLength(10);
+
+        assertThrows(
+            PreferenceBasedSearchNotAllowedException.class,
+            () -> internalCluster().client(nodeMap.get("b").get(0))
+                .prepareSearch()
+                .setSize(0)
+                .setPreference(randomFrom("_local", "_only_nodes:" + nodeInZoneA, "_prefer_nodes:" + nodeInZoneA, customPreference))
```

Comment:
I don't think we should do randomised testing here. Lets assert each of them. Also why should `customPreference `fails

Reply:
Agreed. It is generally an anti-pattern [to use randomization for coverage](https://github.com/opensearch-project/OpenSearch/blob/fe8fd67884160e346def8c8503d09fb4b5eb5363/TESTING.md?plain=1#L440).

Reply:
addressed